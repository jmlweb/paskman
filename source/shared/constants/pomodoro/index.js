export const INTERVAL_TIME = 100;
export const RELOAD_TIMEOUT = 1500;
export const REINIT_TIMEOUT = 1000;
