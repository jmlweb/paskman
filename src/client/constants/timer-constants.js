export const POMODORO_MIN = 25;
export const REST_MIN = 5;
export const POMODORO_STATE = 0;
export const REST_STATE = 1;
export const TIMER_INTERVAL = 500;
export const PAUSE_BETWEEN = false;
