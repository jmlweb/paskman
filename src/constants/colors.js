// @flow
const colors: { [name: string]: string } = {
  sweetPink: '#ffa3a2',
  saintPatrickBlue: '#223177',
}

export default colors;
