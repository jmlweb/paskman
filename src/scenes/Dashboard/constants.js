export default Object.freeze({
  addTaskModalName: 'addTask',
});
