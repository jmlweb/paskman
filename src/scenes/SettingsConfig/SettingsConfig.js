import React from 'react';
import {
  Form,
  FieldSet,
  FormGroup,
  Label,
  OptionsSwitcher,
  RangeSlider,
} from '../../components/Form';
import Heading from '../../components/Heading/Heading';
import Button from '../../components/Button/Button';
import { SettingsPT } from '../../propTypes';

const SettingsConfig = ({
  target,
  pauseBetween,
  confirmEndingTask,
  defaults,
  handleSubmit,
  handleTargetSlider,
  handlePauseBetweenChange,
  handleConfirmEndingTaskChange,
}) => (
  <Form onSubmit={handleSubmit}>
    <Heading>Settings Config</Heading>
    <FieldSet>
      <FormGroup>
        <Label><b>{target.working}</b> minutes working time</Label>
        <RangeSlider
          tooltip={false}
          min={defaults.working.target.min}
          max={defaults.working.target.max}
          step={defaults.working.step}
          value={target.working}
          onChange={handleTargetSlider('working')}
        />
      </FormGroup>
      <FormGroup>
        <Label><b>{target.resting}</b> minutes resting time</Label>
        <RangeSlider
          tooltip={false}
          min={defaults.resting.target.min}
          max={defaults.resting.target.max}
          step={defaults.resting.step}
          value={target.resting}
          onChange={handleTargetSlider('resting')}
        />
      </FormGroup>
      <FormGroup>
        <Label>Pause between pomodoros?</Label>
        <OptionsSwitcher
          options={
            [
              {
                value: true,
                description: 'Yes',
              },
              {
                value: false,
                description: 'No',
              },
            ]
          }
          value={pauseBetween}
          onChange={handlePauseBetweenChange}
        />
      </FormGroup>
      <FormGroup>
        <Label>Require confirm for ending tasks</Label>
        <OptionsSwitcher
          options={
            [
              {
                value: true,
                description: 'Yes',
              },
              {
                value: false,
                description: 'No',
              },
            ]
          }
          value={confirmEndingTask}
          onChange={handleConfirmEndingTaskChange}
        />
      </FormGroup>
    </FieldSet>
    <Button type="submit" color="success" block>Save</Button>
  </Form>
);

SettingsConfig.propTypes = SettingsPT;

export default SettingsConfig;
