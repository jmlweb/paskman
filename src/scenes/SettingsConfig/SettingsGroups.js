import React from 'react';
import PT from 'prop-types';

import { FormGroup, Label, OptionsSwitcher, RangeSlider } from '../../components/Form';

export const buildTimeGroup = ({ target, defaults, handleTargetSlider }) => {
  const TimeGroup = ({ mode }) => (
    <FormGroup>
      <Label>
        <b>{target[mode]}</b> minutes {mode} time
      </Label>
      <RangeSlider
        tooltip={false}
        min={defaults[mode].target.min}
        max={defaults[mode].target.max}
        step={defaults[mode].step}
        value={target[mode]}
        onChange={handleTargetSlider(mode)}
      />
    </FormGroup>
  );
  TimeGroup.propTypes = {
    mode: PT.string.isRequired,
  };
  return TimeGroup;
};

export const BoolGroup = ({ title, value, handleChange }) => (
  <FormGroup>
    <Label>{title}</Label>
    <OptionsSwitcher
      options={[
        {
          value: true,
          description: 'Yes',
        },
        {
          value: false,
          description: 'No',
        },
      ]}
      value={value}
      onChange={handleChange}
    />
  </FormGroup>
);

BoolGroup.propTypes = {
  title: PT.string.isRequired,
  value: PT.bool.isRequired,
  handleChange: PT.func.isRequired,
};
