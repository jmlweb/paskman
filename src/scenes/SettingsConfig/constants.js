const constants = {
  working: {
    step: 5,
    target: {
      min: 5,
      max: 60,
    },
  },
  resting: {
    step: 1,
    target: {
      min: 1,
      max: 15,
    },
  },
};

export default constants;
