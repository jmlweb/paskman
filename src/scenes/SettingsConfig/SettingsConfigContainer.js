import React, { Component } from 'react';
import PT from 'prop-types';
import { connect } from 'react-redux';
import {
  settingsFetchAction,
  settingsSaveAction,
  settingsChangeAction,
} from '../../data/settings/duck';
import Loading from '../../components/Loading/Loading';
import SettingsConfig from './SettingsConfig';
import constants from './constants';

class SettingsConfigContainer extends Component {
  constructor(props) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleTargetSlider = this.handleTargetSlider.bind(this);
    this.handleConfirmEndingTaskChange = this.handleConfirmEndingTaskChange.bind(this);
    this.handlePauseBetweenChange = this.handlePauseBetweenChange.bind(this);
  }
  state = {
    isFetching: true,
  }
  componentDidMount() {
    const { settingsFetch } = this.props;
    settingsFetch();
  }
  componentWillReceiveProps(nextProps) {
    const {
      isFetching,
      target,
      pauseBetween,
      confirmEndingTask,
    } = nextProps;
    this.setState({
      isFetching,
      target,
      pauseBetween,
      confirmEndingTask,
    });
  }
  handleSubmit(e) {
    const { settingsSave } = this.props;
    e.preventDefault();
    settingsSave({
      ...this.state,
    });
  }
  handleTargetSlider(mode) {
    return (value) => {
      this.setState({
        target: {
          ...this.state.target,
          [`${mode}`]: value,
        },
      });
    };
  }
  handleConfirmEndingTaskChange(e) {
    this.setState({
      confirmEndingTask: e.target.value === 'true' && true,
    });
  }
  handlePauseBetweenChange(e) {
    this.setState({
      pauseBetween: e.target.value === 'true' && true,
    });
  }
  render() {
    if (this.props.isFetching || this.state.isFetching) {
      return <Loading text="Loading data..." />;
    }
    return (
      <div>
        {this.props.isSaving && <Loading text="Saving data..." />}
        <SettingsConfig
          target={this.state.target}
          pauseBetween={this.state.pauseBetween}
          confirmEndingTask={this.state.confirmEndingTask}
          handleSubmit={this.handleSubmit}
          handleTargetSlider={this.handleTargetSlider}
          handleConfirmEndingTaskChange={this.handleConfirmEndingTaskChange}
          handlePauseBetweenChange={this.handlePauseBetweenChange}
          defaults={constants}
        />
      </div>
    );
  }
}

SettingsConfigContainer.defaultProps = {
  isFetching: false,
  isSaving: false,
  pauseBetween: false,
  confirmEndingTask: false,
};

SettingsConfigContainer.propTypes = {
  isFetching: PT.bool,
  isSaving: PT.bool,
  target: PT.shape({
    working: PT.number,
    resting: PT.number,
  }).isRequired,
  pauseBetween: PT.bool,
  confirmEndingTask: PT.bool,
  settingsFetch: PT.func.isRequired,
  settingsSave: PT.func.isRequired,
};

export function mapStateToProps(state) {
  return { ...state.data.settings };
}

export const mapDispatchToProps = {
  settingsFetch: settingsFetchAction,
  settingsSave: settingsSaveAction,
  settingsChange: settingsChangeAction,
};

const SettingsConfigConnectedContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(SettingsConfigContainer);

export default SettingsConfigConnectedContainer;
