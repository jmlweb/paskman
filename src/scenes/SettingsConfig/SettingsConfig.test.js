import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import SettingsConfig from './SettingsConfig';
import constants from './constants';

describe('Settings Config', () => {
  const handleSubmit = jest.fn();
  const handleTargetSlider = jest.fn();
  const handlePauseBetweenChange = jest.fn();
  const handleConfirmEndingTaskChange = jest.fn();
  const props = {
    target: {
      working: 25,
      resting: 5,
    },
    pauseBetween: false,
    confirmEndingTask: false,
    defaults: constants,
    handleSubmit,
    handleTargetSlider,
    handlePauseBetweenChange,
    handleConfirmEndingTaskChange,
  };
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <SettingsConfig
        {...props}
      />
    );
  });
  it('Render the component without loading', () => {
    expect(wrapper.find('SettingsConfig').length).toEqual(1);
  });
  it('Handles event', () => {
    wrapper.find('input').first().simulate('change');
    wrapper.find('form').simulate('submit');
    setTimeout(() => {
      expect(handlePauseBetweenChange).toHaveBeenCalled();
      expect(handleSubmit).toHaveBeenCalled();
    }, 500);
  });
  it('Matchs snapshot', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
