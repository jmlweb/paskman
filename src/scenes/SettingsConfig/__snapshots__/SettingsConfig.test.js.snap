// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Config Matchs snapshot 1`] = `
<form
  className="sc-bdVaJa fAoWzh"
  onSubmit={[Function]}
>
  <h1
    className="sc-kAzzGY eLRiub"
  >
    Settings Config
  </h1>
  <fieldset
    className="sc-bwzfXH gSXRtU"
  >
    <div
      className="sc-htpNat QOKog"
    >
      <label
        className="sc-bxivhb fMASpP"
      >
        <b>
          25
        </b>
         minutes working time
      </label>
      <div
        aria-orientation="horizontal"
        aria-valuemax={60}
        aria-valuemin={5}
        aria-valuenow={25}
        className="rangeslider rangeslider-horizontal sc-dnqmqq kGnwsx"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="rangeslider__fill"
          style={
            Object {
              "width": "0px",
            }
          }
        />
        <div
          className="rangeslider__handle"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          style={
            Object {
              "left": "0px",
            }
          }
          tabIndex={0}
        >
          <div
            className="rangeslider__handle-label"
          >
            
          </div>
        </div>
        <ul
          className="rangeslider__labels"
        />
      </div>
    </div>
    <div
      className="sc-htpNat QOKog"
    >
      <label
        className="sc-bxivhb fMASpP"
      >
        <b>
          5
        </b>
         minutes resting time
      </label>
      <div
        aria-orientation="horizontal"
        aria-valuemax={15}
        aria-valuemin={1}
        aria-valuenow={5}
        className="rangeslider rangeslider-horizontal sc-dnqmqq kGnwsx"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="rangeslider__fill"
          style={
            Object {
              "width": "0px",
            }
          }
        />
        <div
          className="rangeslider__handle"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          style={
            Object {
              "left": "0px",
            }
          }
          tabIndex={0}
        >
          <div
            className="rangeslider__handle-label"
          >
            
          </div>
        </div>
        <ul
          className="rangeslider__labels"
        />
      </div>
    </div>
    <div
      className="sc-htpNat QOKog"
    >
      <label
        className="sc-bxivhb fMASpP"
      >
        Pause between pomodoros?
      </label>
      <div
        className="sc-bZQynM kijoyl"
      >
        <label
          className="sc-gzVnrw jpiMBn"
          selected={false}
        >
          <input
            className="sc-htoDjs kCGYta"
            id="id-1"
            onClick={[Function]}
            type="radio"
            value={true}
          />
          <span>
            Yes
          </span>
        </label>
        <label
          className="sc-gzVnrw lhwIJl"
          selected={true}
        >
          <input
            className="sc-htoDjs kCGYta"
            id="id-0"
            onClick={[Function]}
            type="radio"
            value={false}
          />
          <span>
            No
          </span>
        </label>
      </div>
    </div>
    <div
      className="sc-htpNat QOKog"
    >
      <label
        className="sc-bxivhb fMASpP"
      >
        Require confirm for ending tasks
      </label>
      <div
        className="sc-bZQynM kijoyl"
      >
        <label
          className="sc-gzVnrw jpiMBn"
          selected={false}
        >
          <input
            className="sc-htoDjs kCGYta"
            id="id-1"
            onClick={[Function]}
            type="radio"
            value={true}
          />
          <span>
            Yes
          </span>
        </label>
        <label
          className="sc-gzVnrw lhwIJl"
          selected={true}
        >
          <input
            className="sc-htoDjs kCGYta"
            id="id-0"
            onClick={[Function]}
            type="radio"
            value={false}
          />
          <span>
            No
          </span>
        </label>
      </div>
    </div>
  </fieldset>
  <button
    className="sc-chPdSV jlkacU"
    color="success"
    size="md"
    type="submit"
  >
    Save
  </button>
</form>
`;
